### PR TITLE
Api request cloudwatch metrics [MERGED AS #2956]

### PR DESCRIPTION
**Pull request by jamie-pitts**
## What does this change?

This sends CloudWatch metrics for each received request per service (ignoring healthcheck requests). The idea being that it will allow for us to see how many requests per min each service is receiving, if there are any failed requests, and the average length of time each request is taking. The way the cloudwatch metrics are sent has also been slightly modified so that they get aggregated first into a statistic set, this is a lot more useful for count metrics when deployed on multiple instances as it then allows for you to see the average over the amount of instances (e.g. 3 instances each get 100 requests each, you will see an average of 100).

A configurable property has been added `metrics.request.enabled=false` (defaults to false if not present) which can be toggled on a service by service basis if metrics are only desired for a particular service.

Not 100% sure if this is useful for you guys, but thought I'd create a PR none the less!

## Tested?
- [x] locally
- [ ] on TEST

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/lib/config/CommonConfig.scala [hunk 1]
```diff
@@ -50,6 +50,11 @@ trait CommonConfig {
 
   lazy val thrallKinesisStream = properties("thrall.kinesis.stream.name")
 
+  lazy val requestMetricsEnabled: Boolean = properties.getOrElse("metrics.request.enabled", "false").toLowerCase match {
```
Comment (akash1810) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/config/CommonConfig.scala]: we can remove the `match` here in favour of

```suggestion
  lazy val requestMetricsEnabled: Boolean = properties.getOrElse("metrics.request.enabled", "false").toLowerCase == "true"
```

File: common-lib/src/main/scala/com/gu/mediaservice/lib/metrics/CloudWatchMetrics.scala [hunk 2]
```diff
@@ -29,7 +28,7 @@ abstract class CloudWatchMetrics(namespace: String, config: CommonConfig) {
   /** The maximum number of data points to report in one batch.
     * (Each batch costs 1 HTTP request to CloudWatch)
     */
-  val maxChunkSize: Int = 20
+  val maxChunkSize: Int = Int.MaxValue
```
Comment (akash1810) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/metrics/CloudWatchMetrics.scala]: what's the reason for this change?

**Comment by mbarton** [comment]
Hey @jamie-pitts, sorry this has got stuck open for so long. We will rebase it and test in our environment and hopefully merge soon

**Comment by AWare** [comment]
Merged as https://github.com/guardian/grid/pull/2956
